avocado/core/dispatcher.py

```diff
@@ -14,6 +14,7 @@
 
 """Extensions/plugins dispatchers."""
 
+import logging
```

Comment:
OK so the current version does not just ignore the `stdout_claim_by` but it ignores the setting completely:

```
avocado run sleeptenmin.py --json -
JOB ID     : 17c1b71069cff7e818ffe3a7725a5988ffc9a140
JOB LOG    : /home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/job.log
TESTS      : 1
 (1/1) sleeptenmin.py:SleepTenMin.test: -^C
Interrupt requested. Waiting 2 seconds for test to finish (ignoring new Ctrl+C until then)

INTERRUPTED (1.44 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 1
TESTS TIME : 1.44 s
{
    "debuglog": "/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/job.log",
    "errors": 0,
    "failures": 0,
    "job_id": "17c1b71069cff7e818ffe3a7725a5988ffc9a140",
    "pass": 0,
    "skip": 0,
    "tests": [
        {
            "end": 1477918225.697359,
            "fail_reason": "KeyboardInterrupt(): \nRunner error occurred: Interrupted by ctrl+c\nOriginal status: ERROR\n{'traceback': 'Traceback (most recent call last):\\n  File \"/home/medic/Work/Projekty/avocado/avocado/avocado/core/test.py\", line 510, in _run_avocado\\n    raise test_exception\\nTestError: KeyboardInterrupt(): \\n', 'class_name': 'SleepTenMin', 'debugdir': None, 'time_end': 1477918225.697359, 'paused': False, 'time_start': 1477918224.25831, 'basedir': None, 'resultsdir': None, 'time_elapsed': 1.4390490055084229, 'params': <AvocadoParams *: * (['//']),0: /run/* ([])>, 'logdir': '/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/test-results/1-sleeptenmin.py:SleepTenMin.test', 'job_unique_id': '17c1b71069cff7e818ffe3a7725a5988ffc9a140', 'srcdir': '/var/tmp/avocado_eX3RqZ/1-sleeptenmin.py_SleepTenMin.test/src', 'job_logdir': '/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71', 'status': 'ERROR', 'workdir': '/var/tmp/avocado_eX3RqZ/1-sleeptenmin.py_SleepTenMin.test', 'depsdir': None, 'text_output': '13:50:24 DEBUG| PARAMS (key=sleep_cycles, path=*, default=1) => 1\\n13:50:24 DEBUG| PARAMS (key=sleep_length, path=*, default=600) => 600\\n13:50:24 DEBUG| PARAMS (key=sleep_method, path=*, default=builtin) => \\'builtin\\'\\n13:50:24 DEBUG| Sleeping for 600.00 seconds\\n13:50:25 ERROR| \\n13:50:25 ERROR| Reproduced traceback from: /home/medic/Work/Projekty/avocado/avocado/avocado/core/test.py:451\\n13:50:25 ERROR| Traceback (most recent call last):\\n13:50:25 ERROR|   File \"/home/medic/Work/Projekty/avocado/avocado/examples/tests/sleeptenmin.py\", line 31, in test\\n13:50:25 ERROR|     time.sleep(length)\\n13:50:25 ERROR|   File \"/home/medic/Work/Projekty/avocado/avocado/avocado/utils/process.py\", line 409, in signal_handler\\n13:50:25 ERROR|     signal.default_int_handler()\\n13:50:25 ERROR| KeyboardInterrupt\\n13:50:25 ERROR| \\n13:50:25 DEBUG| Local variables:\\n13:50:25 DEBUG|  -> length <type \\'int\\'>: 600\\n13:50:25 DEBUG|  -> cycles <type \\'int\\'>: 1\\n13:50:25 DEBUG|  -> method <type \\'str\\'>: builtin\\n13:50:25 DEBUG|  -> _ <type \\'int\\'>: 0\\n13:50:25 DEBUG|  -> self <class \\'sleeptenmin.SleepTenMin\\'>: 1-sleeptenmin.py:SleepTenMin.test\\n13:50:25 ERROR| ERROR 1-sleeptenmin.py:SleepTenMin.test -> TestError: KeyboardInterrupt(): \\n13:50:25 INFO | \\n', 'running': False, 'sysinfodir': '/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/test-results/1-sleeptenmin.py:SleepTenMin.test/sysinfo', 'logfile': '/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/test-results/1-sleeptenmin.py:SleepTenMin.test/debug.log', 'fail_reason': TestError('KeyboardInterrupt(): ',), 'name': '1-sleeptenmin.py:SleepTenMin.test', 'whiteboard': '', 'fail_class': 'TestError', 'timeout': None, 'paused_msg': ''}",
            "logdir": "/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/test-results/1-sleeptenmin.py:SleepTenMin.test",
            "logfile": "/home/medic/avocado/job-results/job-2016-10-31T13.50-17c1b71/test-results/1-sleeptenmin.py:SleepTenMin.test/debug.log",
            "start": 1477918224.25831,
            "status": "INTERRUPTED",
            "test": "1-sleeptenmin.py:SleepTenMin.test",
            "time": 1.4390490055084229,
            "url": "1-sleeptenmin.py:SleepTenMin.test",
            "whiteboard": ""
        }
    ],
    "time": 1.4390490055084229,
    "total": 1
}
```

Well you need to address this issue before the final version. I'd go with the silent `avocado.app` by default and enabling it only when `HumanResult` is used. When `json` (or other) plugin is used they should use a different stream and keep the `avocado.app` disabled so any messages from let's say `html` result is ignored (unless `avocado.app` is enabled).

It's actually even simpler than what I had originally in mind. Basically we just need to check whether there is an output claimed by anything, if not then enable `avocado.app` and provide the default output. Otherwise keep it disabled and leave it on the plugins to enable their output to stdout.